src/main/kotlin/xyz/crossward/controller/GroupController.kt

```diff
@@ -0,0 +1,50 @@
+package xyz.crossward.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.*
+import xyz.crossward.entities.Group
+import xyz.crossward.service.GroupService
+import java.net.URLDecoder
+
+@RestController
+@RequestMapping("/api")
+class GroupController(
+        var service: GroupService
+) {
+
+    @PostMapping("/groups/telegram")
+    @ResponseStatus(HttpStatus.CREATED)
+    fun createTelegramGroup(@RequestBody group: Group): Group {
+        return service.createTelegramGroup(group)
+    }
+
+    @PostMapping("/groups/website")
+    @ResponseStatus(HttpStatus.CREATED)
+    fun createGroup(@RequestBody group: Group): Group {
+        return service.createWebsiteGroup(group)
+    }
+
+    @GetMapping("/groups/ids/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional(readOnly = true)
+    fun getGroupById(@PathVariable("id") id: String): ResponseEntity<Group> {
```

Comment:
We should move to using query params instead of path variables. So remove the `{id}` from the GET annotation and change `@PathVariable` to `@RequestParam`